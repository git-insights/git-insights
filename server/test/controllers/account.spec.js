jest.mock("../../models");
const models = require("../../models");
const { mockRequest, mockResponse } = require("../utils/interceptor.js");
const { getUser } = require("../../src/controllers/account.js");

describe("Controller 'getUser' ", () => {
  test("should respond with an object with `null` user, when user doesn't exist in request.", async () => {
    models.User = {
      findOne: jest.fn()
    };

    const req = mockRequest();
    const res = mockResponse();
    await getUser(req, res);

    expect(res.json).toHaveBeenCalledTimes(1);
    expect(res.json).toMatchSnapshot();
    expect(models.User.findOne).toHaveBeenCalledTimes(0);
  });

  test("should respond with user model, when user exists in request.", async () => {
    models.User = {
      findOne: jest.fn()
    };

    models.User.findOne.mockReturnValue({ id: "USER_ID_TEST" });

    const req = mockRequest("../fixtures/get_user.json");
    const res = mockResponse();

    await getUser(req, res);

    expect(models.User.findOne).toHaveBeenCalledTimes(1);
    expect(models.User.findOne).toMatchSnapshot()

    expect(res.json).toHaveBeenCalledTimes(1);
    expect(res.json).toMatchSnapshot();
  });
});
