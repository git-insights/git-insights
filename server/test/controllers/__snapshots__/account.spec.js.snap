// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller 'getUser'  should respond with an object with \`null\` user, when user doesn't exist in request. 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "user": null,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "json": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "user": null,
              },
            ],
          ],
          "results": [Circular],
        },
        "send": [MockFunction],
        "status": [MockFunction],
      },
    },
  ],
}
`;

exports[`Controller 'getUser'  should respond with user model, when user exists in request. 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "include": Object {
          "as": "repos",
          "model": [MockFunction],
        },
        "where": Object {
          "id": "USER_ID_TEST",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "id": "USER_ID_TEST",
      },
    },
  ],
}
`;

exports[`Controller 'getUser'  should respond with user model, when user exists in request. 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "user": Object {
          "id": "USER_ID_TEST",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "json": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "user": Object {
                  "id": "USER_ID_TEST",
                },
              },
            ],
          ],
          "results": [Circular],
        },
        "send": [MockFunction],
        "status": [MockFunction],
      },
    },
  ],
}
`;
